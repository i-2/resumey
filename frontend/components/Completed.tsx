import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Button, Flex, Heading, Text } from '@chakra-ui/react';
import * as actionCreators from '../actions';

export const _Completed = (props: any) => {
    const { hasPublished, urlLink } = props.publish;
    const onPublish = () => props.tryPublish();
    return (
        <Flex width="100%" direction="column" justify="center" borderWidth="1px" borderRadius="lg" padding="5%" marginTop="2%">
            <Flex width="100%" direction="row" justify="center" padding="5%" marginTop="2%"><Heading>  Almost Complete! </Heading> </Flex>
            {
                !hasPublished ? <Flex width="100%" direction="row" justify="center" padding="5%" marginTop="2%"><Button colorScheme="teal" onClick={onPublish}> Publish Now </Button></Flex> :
                    <Flex width="100%" direction="row" justify="center" padding="5%" marginTop="2%">
                        <a href={urlLink}>
                            <Text>{urlLink}</Text>
                        </a>
                    </Flex>
            }
        </Flex>
    )
}

const mapStateToProps = (state: any) => ({
    publish: state.publish
})
const mapDispatchToProps = (dispatch: any) => bindActionCreators(actionCreators, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(_Completed);