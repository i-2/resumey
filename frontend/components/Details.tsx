import React from 'react';
import {
    Flex, Button, FormControl, FormLabel, Input, FormErrorMessage, Wrap, WrapItem
} from '@chakra-ui/react';
import {
    Field,
    Formik
} from 'formik';
import * as Yup from 'yup';

export interface ValueProps {
    name: string,
    title: string,
    validate?: any
}

export interface DetailProps {
    schema: ValueProps[],
    initValues: any,
    onNext: () => any,
    submitValues: (value: any) => any
    onPrevious?: () => any
}

function getValues(schema: ValueProps[]) {
    let values: any = {};
    for (let value of schema) {
        values[value.name] = "";
    }
    return values;
}

function getValidation(schema: ValueProps[]) {
    let values: any = {};
    for (let value of schema) {
        values[value.name] = value.validate;
    }
    return Yup.object().shape(values);
}


export function renderField(values: ValueProps[]) {
    return (<Wrap width="100%" borderWidth="1px" borderRadius="lg" padding="5%" marginTop="2%">
        {
            values.map((value: any, i: number) =>
                <WrapItem key={i} width="45%" margin="10px">
                    <Field name={value.name}>
                        {({ field, form }: any) => (
                            <FormControl isInvalid={form.errors[value.name] && form.touched[value.name]}>
                                <FormLabel htmlFor={value[value.name]}> {value.title} </FormLabel>
                                <Input {...field}
                                    id={value[value.name]}
                                    placeholder={value.title}
                                />
                                <FormErrorMessage>{form.errors[value.name]}</FormErrorMessage>
                            </FormControl>
                        )}
                    </Field>
                </WrapItem>)}
    </Wrap>)
}



export const Details = (props: DetailProps) => {

    return (<Flex width="100%" direction="row" justify="center">
        <Flex width="100%" direction="column" justify="center">
            <Formik
                initialValues={props.initValues}
                validationSchema={getValidation(props.schema)}
                onSubmit={(values, actions) => {
                    props.submitValues(values);
                    props.onNext();
                }}>
                {
                    (innerProps: any) => (
                        <form style={{ width: "100%" }} onSubmit={innerProps.handleSubmit}>
                            <Flex width="100%" direction="row" justify="space-between">
                                {renderField(props.schema)}
                            </Flex>
                            {
                                props.onPrevious ? <Button
                                    mt={4}
                                    colorScheme="teal"
                                    type="submit"
                                    m="5"
                                    onClick={props.onPrevious}
                                >
                                    Previous </Button> : <div></div>
                            }
                            <Button
                                mt={4}
                                colorScheme="teal"
                                type="submit"
                                m="5"
                            >
                                Next
                            </Button>


                        </form>
                    )
                }
            </Formik>
        </Flex>
    </Flex >)
}