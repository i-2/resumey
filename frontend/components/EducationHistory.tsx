import React from 'react';
import { ArrayForm } from './chakraForms/ArrayForm';
import * as Yup from 'yup';


const EDUCATION_FIELDS = [
    { name: "school", title: "School", placeholder: "Enter the name of school", "type": "input", "validate": Yup.string().required() },
    { name: "degree", title: "Degree", placeholder: "Enter the name of degree", "type": "input", "validate": Yup.string().required() },
    { name: "start", title: "Start Date", placeholder: "Enter the start date", "type": "date", "validate": Yup.date().required() },
    { name: "end", title: "End Date", placeholder: "Enter the end date", "type": "date", "validate": Yup.date().required() },
    { name: "city", title: "City", placeholder: "Enter the place of study", "type": "input", "validate": Yup.string().required()},
    { name: "country", title: "Country", placeholder: "Enter the country of country", "type": "input", "validate": Yup.string().required()},
    { name: "description", title: "Description", placeholder: "Care to explain ?", "type": "text", "validate": Yup.string().required() }
]


export const EducationHistory = (props: any) => <ArrayForm
    name="education"
    fields={EDUCATION_FIELDS}
    initValues={{ "education": props.education }}
    onPrevious={props.onPrevious}
    onAddMore={props.onAddMore}
    onSubmit={props.onSave}
    onNext={props.onNext}
/>