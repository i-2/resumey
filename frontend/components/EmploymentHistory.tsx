import React from 'react';
import { ArrayForm } from './chakraForms/ArrayForm';


const EMPLOYMENT_FIELDS = [
    { name: "employer", title: "Employer", placeholder: "Enter the name of employer", "type": "input" },
    { name: "designation", title: "Designation", placeholder: "Enter the designation", "type": "input" },
    { name: "start", title: "Start Date", placeholder: "Enter the start date", "type": "date" },
    { name: "end", title: "End Date", placeholder: "Enter the end date", "type": "date" },
    { name: "summary", title: "Summary", placeholder: "Summary", "type": "text" }
]


export const EmploymentHistory = (props: any) => <ArrayForm
    name="history"
    fields={EMPLOYMENT_FIELDS}
    initValues={{"history": props.history}}
    onPrevious={props.onPrevious}
    onAddMore={props.onAddMore}
/>