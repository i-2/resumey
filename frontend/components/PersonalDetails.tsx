import React from 'react';
import {
    Flex, Button, FormControl, FormLabel, Input, FormErrorMessage, Text, Wrap, WrapItem
} from '@chakra-ui/react';
import {
    Form,
    Field,
    Formik,
} from 'formik';


export interface ValueProps {
    name: string,
    title: string
}

export interface PersonalDetailProps {
    schema: ValueProps[],
    onNext: () => any;
}


export function renderField(values: ValueProps[]) {
    return (<Wrap width="100%">
        {values.map((value: any, i: number) => <WrapItem key={i} width="45%" margin="10px"><Field name={value.name}>
            {({ field, form }: any) => (
                <FormControl isInvalid={form.errors.name && form.touched.name}>
                    <FormLabel htmlFor={value.name}> {value.title} </FormLabel>
                    <Input {...field} id={value.name} placeholder={value.title} />
                    <FormErrorMessage>{form.errors.name}</FormErrorMessage>
                </FormControl>
            )}
        </Field></WrapItem>)}
    </Wrap>)
}



export const PersonalDetails = (props: PersonalDetailProps) => (
    <Flex width="100%" direction="row" justify="center">
        <Formik
            initialValues={{}}
            onSubmit={(values, { setSubmitting }) => { }}>
            <Flex width="100%" direction="column" justify="center">
                <Form style={{ width: "100%" }}>
                    <Flex width="100%" direction="row" justify="space-between">
                        {renderField(props.schema)}
                    </Flex>
                    <Button
                        mt={4}
                        colorScheme="teal"
                        isLoading={false}
                        type="submit"
                        m="5"
                        onClick={props.onNext}
                    >
                        Next
                        </Button>

                </Form>
            </Flex>
        </Formik>
    </Flex>


)