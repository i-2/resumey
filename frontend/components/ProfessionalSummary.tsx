import React from 'react';
import {
    Flex, FormControl, FormLabel, Textarea, FormErrorMessage, Button
} from '@chakra-ui/react';
import {
    Field,
    Formik,
} from 'formik';
import * as Yup from 'yup';

const SummarySchema = Yup.object().shape(
    { "summary": Yup.string().required() }
)

export interface ProfessionalSummaryProps {
    name: string,
    title: string,
    initValues: any,
    onNext: () => any,
    onPrev: () => any,
    submitValues: (values: any) => any
}

export function ProfessionalSummary(props: ProfessionalSummaryProps) {
    const { name, title } = props;
    return (

        <Flex width="100%" direction="row" justify="center">
            <Formik
                initialValues={props.initValues}
                validationSchema={SummarySchema}
                onSubmit={(values, actions) => {
                    props.onNext();
                    props.submitValues(values);
                }}>
                {
                    (innerProps: any) => (
                        <Flex width="100%" direction="column" justify="center">
                            <form style={{ width: "100%" }} onSubmit={innerProps.handleSubmit}>
                                <Field name={name}>
                                    {({ field, form }: any) => {

                                        return (

                                            <FormControl isInvalid={form.errors.summary && form.touched.summary}>
                                                <FormLabel htmlFor={name}> {title} </FormLabel>
                                                <Textarea placeholder="Enter your professional summary"  {...field} />
                                                <FormErrorMessage>{form.errors.summary}</FormErrorMessage>
                                            </FormControl>

                                        )
                                    }
                                    }
                                </Field>
                                <Button
                                    mt={4}
                                    colorScheme="teal"
                                    isLoading={false}
                                    onClick={props.onPrev}
                                    m="5"
                                >
                                    Previous
                                </Button>
                                <Button
                                    mt={4}
                                    colorScheme="teal"
                                    isLoading={false}
                                    type="submit"
                                    m="5"
                                >
                                    Next
                                </Button>
                            </form>
                        </Flex>
                    )
                }

            </Formik>
        </Flex>

    );
}