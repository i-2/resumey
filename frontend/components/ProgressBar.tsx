import React from 'react';
import { Box, Progress, Flex, Spacer } from '@chakra-ui/react';
import { QuestionOutlineIcon } from '@chakra-ui/icons';

export interface ProgressBarProps {
    value: number,
}

export const ProgressBar = (props: ProgressBarProps) => (
    <Box width="inherit" borderRadius="50" m="5" centerContent>
        <Flex>
            <Box p="1">
                Completion : {props.value}%
            </Box>
            <Spacer />
            <Box p="1" as="button" >
                <QuestionOutlineIcon w={6} h={6} />
            </Box>
        </Flex>
        <Progress value={props.value} size="xs" colorScheme="green" isAnimated />
    </Box>
)