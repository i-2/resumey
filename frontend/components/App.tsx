import React from 'react';
import { Flex, Text } from '@chakra-ui/react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actionCreators from '../actions';
import { Details } from './Details';
import { ProfessionalSummary } from './ProfessionalSummary';
import { Wizard } from './Wizard';
import * as Yup from 'yup';

const PERSONAL_DETAIL_SCHEMA: any[] = [
    { "title": "First Name", "name": "firstName", "validate": Yup.string().required() },
    { "title": "Last Name", "name": "lastName", "validate": Yup.string().required() },
    { "title": "Email", "name": "email", "validate": Yup.string().email().required() },
    { "title": "Phone Number", "name": "phone", "validate": Yup.string().required() },
    { "title": "Country", "name": "country", "validate": Yup.string().required() },
    { "title": "City", "name": "city", "validate": Yup.string().required() },
    { "title": "Address", "name": "address", "validate": Yup.string().required() },
    { "title": "Postal Code", "name": "pinCode", "validate": Yup.string().required() }
]


export const _App = (props: any) => (
    <Flex direction="column"
        align="center"
        maxW={{ xl: "1200px" }}
        m="0 auto">
        <Flex w="100%" justify="center">
            <Text fontSize="3xl"> Resume Builder </Text>
        </Flex>
        <Wizard tabs={
            [
                {
                    "title": "Personal Details", "component": <Details
                        schema={PERSONAL_DETAIL_SCHEMA}
                        values={props.values}
                        onNext={() => props.updateWizard(1)}
                        onDetailChange={(name, value) => {
                            props.updatePersonalDetail(name, value);
                        }}
                    />
                },
                { "title": "Professional Summary", "component": <ProfessionalSummary title="Summary" name="summary" onNext={() => props.updateWizard(2)} onPrev={() => props.updateWizard(0)} /> }
            ]
        } count={props.count} />
    </Flex>
)

const mapStateToProps = (state: any) => ({ count: state.wizard.windowCount, values: state.personalDetail })
const mapDispatchToProps = (dispatch: any) => bindActionCreators(actionCreators, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(_App);