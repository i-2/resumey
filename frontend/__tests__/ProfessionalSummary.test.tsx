import * as  React from 'react';
import * as renderer from 'react-test-renderer';
import {ProfessionalSummary } from '../components/ProfessionalSummary';

it('renders completed correctly', () => {
  const tree = renderer
    .create(<ProfessionalSummary />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});