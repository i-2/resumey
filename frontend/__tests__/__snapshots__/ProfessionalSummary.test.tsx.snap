// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders completed correctly 1`] = `
<div
  className="css-2l42up"
>
  <div
    className="css-1bgjfgj"
  >
    <form
      onSubmit={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <label
          className="chakra-form__label css-1ig2arj"
          htmlFor=""
          id="field-1-label"
        >
           
          
           
        </label>
        <textarea
          aria-describedby="field-1-feedback"
          className="chakra-textarea css-0"
          disabled={false}
          id="field-1"
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter your professional summary"
          readOnly={false}
          required={false}
          value={Object {}}
        />
      </div>
      <button
        className="chakra-button css-1bgvtth"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Previous
      </button>
      <button
        className="chakra-button css-1bgvtth"
        disabled={false}
        type="submit"
      >
        Next
      </button>
    </form>
  </div>
</div>
`;
